src/hooks/core/profile/useProfileHook.ts

```diff
@@ -7,7 +7,9 @@ import {ProfileContext} from '../../../../types/context/profilecontext.types';
 
 const useProfileHook = () => {
   // TODO: Merge this with useUserAuthHook and delete this hook
-  const [profile] = React.useContext(Context).profile;
+  const contextValue = React.useContext(Context);
+  const profile = contextValue ? contextValue.profile : null;
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces a null check for `contextValue` which is a good practice to avoid potential null reference errors. However, the type casting of `profile` to `ProfileContext` should be done after checking if `profile` is not null.
```diff
-  const profile = contextValue ? contextValue.profile : null;
-  const profileContext = profile as ProfileContext;
+  const profile = contextValue?.profile;
+  const profileContext = profile ? profile as ProfileContext : null;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->